diff --git a/browser/net/brave_proxying_url_loader_factory.cc b/browser/net/brave_proxying_url_loader_factory.cc
--- a/browser/net/brave_proxying_url_loader_factory.cc
+++ b/browser/net/brave_proxying_url_loader_factory.cc
@@ -54,8 +54,9 @@ std::uint32_t InitiatorProcessId(
   }
   content::RenderProcessHost* const render_process =
       content::RenderProcessHost::FromID(render_frame_token.child_id);
-  return render_process
-             ? static_cast<std::uint32_t>(render_process->GetProcess().Pid())
-             : 0;
+  if (!render_process || !render_process->GetProcess().IsValid()) {
+    return 0;
+  }
+  return static_cast<std::uint32_t>(render_process->GetProcess().Pid());
 }
 
