diff --git a/renderer/brave_content_renderer_client.cc b/renderer/brave_content_renderer_client.cc
index 05261e5..30fb7be 100644
--- a/renderer/brave_content_renderer_client.cc
+++ b/renderer/brave_content_renderer_client.cc
@@ -8,6 +8,7 @@
 #include <algorithm>
 #include <utility>
 
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "brave/components/ai_chat/core/common/buildflags/buildflags.h"
 #include "brave/components/brave_news/renderer/rss_link_reader.h"
@@ -157,6 +158,15 @@ void BraveContentRendererClient::RenderThreadStarted() {
       brave::SafeBuiltins::CreateV8Extension());
 }
 
+bool BraveContentRendererClient::ShouldReportDetailedMessageForSource(
+    const std::u16string& source) {
+  static const bool enabled =
+      base::CommandLine::ForCurrentProcess()->HasSwitch("reb-native-console");
+  return enabled ||
+         ChromeContentRendererClient::ShouldReportDetailedMessageForSource(
+             source);
+}
+
 void BraveContentRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
   ChromeContentRendererClient::RenderFrameCreated(render_frame);
diff --git a/renderer/brave_content_renderer_client.h b/renderer/brave_content_renderer_client.h
index 9e15040..b2153bd 100644
--- a/renderer/brave_content_renderer_client.h
+++ b/renderer/brave_content_renderer_client.h
@@ -31,6 +31,8 @@ class BraveContentRendererClient : public ChromeContentRendererClient {
   void RenderThreadStarted() override;
   void SetRuntimeFeaturesDefaultsBeforeBlinkInitialization() override;
   void RenderFrameCreated(content::RenderFrame* render_frame) override;
+  bool ShouldReportDetailedMessageForSource(
+      const std::u16string& source) override;
   std::unique_ptr<media::KeySystemSupportRegistration> GetSupportedKeySystems(
       content::RenderFrame* render_frame,
       media::GetSupportedKeySystemsCB cb) override;
